src/site/_collections/authors.js

```diff
@@ -38,89 +37,83 @@ const findAuthorsPosts = (posts) => {
  * Returns all authors with their posts.
  *
- * @param {any} collections Eleventy collection object
+ * @param {any} [collections] Eleventy collection object
  * @return {Object.<string, Author>}
  */
 module.exports = (collections) => {
   // Get all posts and sort them
-  const posts = collections
-    .getFilteredByGlob('**/*.md')
-    .filter(livePosts)
-    .sort((a, b) => b.date - a.date);
+  let posts = [];
+
+  if (collections) {
+    posts = collections
+      .getFilteredByGlob('**/*.md')
+      .filter(livePosts)
+      .sort((a, b) => b.date - a.date);
+  }
 
   const authorsPosts = findAuthorsPosts(posts);
 
   /** @constant @type {Object.<string, Author>} @default */
   const authors = {};
 
-  Object.values(contributors)
-    .sort((a, b) => a.title.localeCompare(b.title))
-    .forEach((author) => {
-      // This updates the shared contributors object with meta information and is safe to be called multiple times.
-      author.url = path.join('/en', author.href);
-      author.data = {
-        title: author.title,
-        subhead: author.description,
-      };
+  Object.keys(authorsData).forEach((key) => {
+    const author = {...authorsData[key]};
+    author.key = key;
 
-      author.elements = authorsPosts.has(author.key)
-        ? authorsPosts.get(author.key)
-        : [];
+    // Generate the author's name out of valid given/family parts. This
+    // allows our authors to just have a single name.
+    const parts = [author.name.given, author.name.family].filter(
+      (s) => s && s.length,
+    );
+    author.title = parts.join(' ');
+    author.href = `/authors/${key}/`;
+    author.description =
+      author.description && author.description.en
+        ? author.description.en
+        : `Our latest news, updates, and stories by ${author.title}.`;
+    author.url = path.join('/en', author.href);
+    author.data = {
+      title: author.title,
+      subhead: author.description,
+    };
 
-      // If the author doesn't have any posts, use their Twitter profile.
-      if (author.elements.length === 0) {
-        if (!author.twitter) {
-          // If we're screenshot testing just ignore it.
-          if (process.env.PERCY) {
-            return;
-          }
+    author.elements = authorsPosts.has(key) ? authorsPosts.get(key) : [];
 
-          // posts.length might be empty if we're generating partials so only
-          // log a warning if it has a value and we're doing a regular build.
-          // Note this is checking for _all_ posts and not just the posts
-          // by this specific author.
-          if (posts.length) {
-            console.warn(
-              `author ${
-                author.title
-              } has no posts and no social: ${JSON.stringify(author)}\n`,
-            );
-          }
-        } else {
-          author.href = `https://twitter.com/${author.twitter}`;
+    // If the author doesn't have any posts, use their Twitter profile.
+    if (author.elements.length === 0) {
```

Comment:
Would it be possible to pull @samthor's changes from https://github.com/GoogleChrome/web.dev/pull/3330 into this PR?

Reply:
Or we could land this in do a separate PR if you'd like to reduce the number of changes. Up to you.

Reply:
@samthor would you mind if I merged your PR into mine?

Reply:
Go ahead.